internal/services/nginx/nginx_configuration_resource.go

```diff
@@ -253,21 +253,34 @@ func (m ConfigurationResource) Read() sdk.ResourceFunc {
 				}
 
 				if files := prop.Files; files != nil {
+					configs := []ConfigFile{}
 					for _, file := range *files {
-						output.ConfigFile = append(output.ConfigFile, ConfigFile{
-							Content:     pointer.ToString(file.Content),
-							VirtualPath: pointer.ToString(file.VirtualPath),
-						})
+						if pointer.From(file.Content) != "" {
+							configs = append(configs, ConfigFile{
+								Content:     pointer.ToString(file.Content),
+								VirtualPath: pointer.ToString(file.VirtualPath),
+							})
+						}
+					}
+					if len(configs) > 0 {
+						output.ConfigFile = configs
 					}
 				}
 
 				// GET does not return protected files
 				if files := prop.ProtectedFiles; files != nil {
+					configs := []ProtectedFile{}
 					for _, file := range *files {
-						output.ProtectedFile = append(output.ProtectedFile, ProtectedFile{
-							Content:     pointer.ToString(file.Content),
-							VirtualPath: pointer.ToString(file.VirtualPath),
-						})
+						if pointer.From(file.Content) != "" {
```

Comment:
@wuxu92 I am curious to know why this issue is occurring. I tried looking into the provider code and here's what I have:

- We are using API version 2024-06-01.
- A while back, the service never returned `protectedFiles` and so we used to get the information in the `Read()` method by decoding the information from the state and that avoided issues with diffs on repeated plans.
- Now the service is returning `protectedFiles` but without `content`.
- The comment and logic behind `// GET does not return protected files` is no longer valid.
- If I use `curl` to access the configuration while specifying the API version, I get the following back:
```shell
    "protectedFiles": [
      {
        "content": null,
        "virtualPath": "/var/www/index.html"
      }
    ],
```

Keeping this information in mind, I built a custom provider with `content` of the protected file not marked as a `Sensitive` field to observe the diffs on repeated plans and I get the following:

```shell
Terraform will perform the following actions:

  # azurerm_nginx_configuration.example will be updated in-place
  ~ resource "azurerm_nginx_configuration" "example" {
        id                  = "assume-the-correct-resourceID
        # (2 unchanged attributes hidden)

      - protected_file {
          - virtual_path = "/etc/nginx/site/api.conf" -> null
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

Why is the provider trying to nullify the `virtualPath`?

Reply:
Tagging @katbyte as well on the question^

Reply:
got it. thanks for explaining.